### PR TITLE
EIP1-3823 Add secure endpoint to generate an Anonymous Elector Document (AED)

### PR DESCRIPTION
**Pull request by Neil-Massey**
Example AED

image

Mainly the mapper from REST API to DTO, Controller Method and Integration Test with few minor fixes and updates.

### CODE REVIEW

File: src/main/kotlin/uk/gov/dluhc/printapi/service/aed/AedPdfTemplateDetailsFactory.kt [hunk 1]
```diff
@@ -32,7 +32,7 @@ class AedPdfTemplateDetailsFactory(
         printRequest: AedPrintRequest
     ): Map<String, String> {
         return mapOf(
-            pdfTemplateProperties.welsh.placeholder.electoralRollNumber to printRequest.electoralRollNumber,
+            pdfTemplateProperties.welsh.placeholder.electoralRollNumber to printRequest.electoralRollNumber.uppercase(),
```
Comment (matt-wills-valtech) [inline src/main/kotlin/uk/gov/dluhc/printapi/service/aed/AedPdfTemplateDetailsFactory.kt:35]: 👍

**Review by matt-wills-valtech — APPROVED** [review]

File: src/main/kotlin/uk/gov/dluhc/printapi/service/aed/AedPdfTemplateDetailsFactory.kt (hunk 1 above)
Comment (Neil-Massey) [reply src/main/kotlin/uk/gov/dluhc/printapi/service/aed/AedPdfTemplateDetailsFactory.kt:35]: To be honest I don't know that the value is case-insensitive and don't generally like to modify user provided entries.

File: src/main/resources/openapi/PrintAPIs.yaml [hunk 2]
```diff
@@ -875,7 +875,7 @@ components:
         - applicationReference
         - electoralRollNumber
         - certificateLanguage
-        - supportingInfoFormat
+        - supportingInformationFormat
```
Comment (vishalgupta-valtech) [inline src/main/resources/openapi/PrintAPIs.yaml:878]: Isnt `supportingInformationFormat` optional in dto that you created? How does that work

File: src/test/kotlin/uk/gov/dluhc/printapi/rest/GenerateAnonymousElectorDocumentIntegrationTest.kt [hunk 3]
```diff
@@ -0,0 +1,265 @@
...
+        wireMockService.stubEroManagementGetEroByGssCodeNoMatch(GSS_CODE)
+        val requestBody = buildGenerateAnonymousElectorDocumentRequest(gssCode = GSS_CODE)
+
+        // When
+        val response = webTestClient.post()
+            .uri(
+                URI_TEMPLATE,
+                ERO_ID
+            )
+            .bearerToken(
```
Comment (vishalgupta-valtech) [inline src/test/kotlin/uk/gov/dluhc/printapi/rest/GenerateAnonymousElectorDocumentIntegrationTest.kt:111]: Not in the scope of current PR however we could introduce similar functions that we currently introduced in vca. 
`.bearerToken(getVCAdminBearerToken(eroId = ERO_ID))`

```
fun getAnonymousBearerToken(eroId: String = ERO_1234, userName: String = "an-ero-user1@$eroId.gov.uk") =
    getBearerToken(groups = listOf("ero-$eroId", "ero-vc-anonymous-admin-$eroId"), email = userName)

fun getVCAdminBearerToken(eroId: String = ERO_1234, userName: String = "an-ero-user2@$eroId.gov.uk") =
    getBearerToken(groups = listOf("ero-$eroId", "ero-vc-admin-$eroId"), email = userName)
```

**Review by vishalgupta-valtech — APPROVED** [review]
LGTM with one minor query

**Pull request merged**